20 メモ一覧取得api作成

soichiro1357533 opened the pull request:
## 関連Issue
<!-- #の後ろに対応するissue番号を書く -->
Closes #20 

## やったこと
メモ一覧を取得するapiを作成
## 動作確認
<!-- テスト内容など -->
- [ ] [/api/auth/me](http://localhost:8080/swagger-ui/index.html#/Authentication/getCurrentUser)ここで取得したIDを直接データベースに書き込み一覧が表示されるかテストしました
<img width="1205" height="894" alt="image" src="https://github.com/user-attachments/assets/6079a52c-c526-4978-9370-5c74202dc89c" />


## 参考にした資料

soichiro1357533 committed:
20 メモ一覧取得api作成

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+    // ユースケースは Bean 登録せず、このレイヤーで組み立てる
+    voiceMemoRepository: VoiceMemoRepository,
+) {
+    // Bean 化しないことでユースケースをフレームワーク非依存に保つ
```
この辺のBeanコメントは削除して欲しいです。

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+        ],
+    )
+    suspend fun listMemos(
+        @AuthenticationPrincipal userId: UUID?,
```
voicecontrollerの方でこの@AuthenticationPrincipalを使用せずにuserIdを取得している実装をしていたので、issueを発行しました。 @soichiro1357533 がやってくれている実装の方が正しいです。
https://github.com/assari-harassment/voicebooklm-backend/issues/53
このコメント確認したらresolveで閉じてもらって構わないです。

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+    // ユースケースは Bean 登録せず、このレイヤーで組み立てる
+    voiceMemoRepository: VoiceMemoRepository,
+) {
+    // Bean 化しないことでユースケースをフレームワーク非依存に保つ
+    private val listMemosUseCase = ListMemosUseCase(voiceMemoRepository)
+
+    @GetMapping("/memos")
+    @Operation(
+        summary = "メモ一覧取得",
+        description = "検索は未対応。認証ユーザーのメモを新しい順で取得する。",
+        responses = [
+            ApiResponse(
+                responseCode = "200",
+                description = "取得成功",
+                content = [Content(schema = Schema(implementation = ListMemosResponse::class))],
+            ),
+            ApiResponse(
+                responseCode = "401",
+                description = "認証エラー",
+                content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+            ),
+        ],
+    )
+    suspend fun listMemos(
+        @AuthenticationPrincipal userId: UUID?,
+    ): ResponseEntity<ListMemosResponse> {
+        userId ?: return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build()
+        val result = listMemosUseCase.execute(ListMemosInput(userId))
+        return ResponseEntity.ok(ListMemosResponse.from(result))
+    }
+}
```
テストは作成して欲しいです。

soichiro1357533 replied:
MemoController 向けのコントローラーテストを作成し、認証エラーの 401 と空リスト返却を含めて検証しています

junhat6 commented:
メモ：
folder_pathカラムを追加したら、こちら変更しないといけないので、folder_pathのissue早めに取り掛かった方が良さそう

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+data class MemoSummaryResponse(
+    val memoId: UUID,
+    val title: String,
+    val content: String,
```
contentはテキスト量が多いので、一覧では返さないでください。
title 
tags
createdatがフロントエンドの一覧では表示されます。
<img width="300" height="229" alt="image" src="https://github.com/user-attachments/assets/e4d8c7cd-9515-4d52-8063-fad96e9fe9e9" />
updatedAtとtranscriptionStatusとformattingStatusも返して欲しいです。
updatedAtはあとで仕様の変更で必要になるかもなんで、返しといて欲しいです。
transcriptionStatusとformattingStatusはフロントエンド側で処理中のものを表示するのかどうかの仕様が決まっていないので、返しておいてください。
仕様としてはformattingStatusが完了のものをフロント側で表示したいとは考えていますが、処理失敗したものにどうやってアクセスするの問題があるので、返しておいてください。

soichiro1357533 replied:
一覧APIのレスポンスに content を含めず、title/tags/createdAt/updatedAt/transcriptionStatus/formattingStatus を返してます

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+/**
+ * メモ一覧の要素
+ */
+data class MemoSummaryResponse(
```
VoiceControllerではそうしてないんですけど（すみませんリファクターのissue立てます。）
responseとrequestは違うファイルに定義して欲しいで。
理由としては後々このコントローラーに詳細取得とか色々実装すると思うんですが、request responseを別ファイルに定義しておかないとコントローラーが１０００行にも行っちゃう恐れがあるからです。
あと、シンプルにユースケースを実装するときにファイルがrequest responseが分かれていると探しやすくて、実装しやすいからです。

junhat6 replied:
https://github.com/assari-harassment/voicebooklm-backend/issues/54
issue作成しました！

junhat6 replied:
summaryは要約という意味で持ってる人が多いと思うので、MemoSummaryResponseではなくて違う名前にして欲しいです。
（僕にはパッと思いつかない　すみません）

junhat6 committed:
正常なオニオンアーキテクチャへのリファクタリング #48

# Conflicts:
#	src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt

junhat6 committed:
VoiceController.ktレスポンスクラスを別ファイルに切り出し #48

junhat6 committed:
冗長なコメント削除 #48

junhat6 committed:
例外クラスをErrorCode enum + DomainExceptionに統一 #55

soichiro1357533 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,99 @@
+                MemoSummaryResponse(
+                    memoId = memo.id,
+                    title = memo.title ?: "",
+                    content = memo.content ?: "",
```
整形未完了のメモを空文字に潰さず、null を返すように変更しました

soichiro1357533 committed:
レビュー対応と依存関係が少し間違っていたので修正 #20

junhat6 committed:
コンフリクト対応 #55

junhat6 committed:
Merge pull request #56 from assari-harassment/feature/55

junhat6 commented on src/main/kotlin/com/assari/voicebooklm/presentation/controller/memo/MemoController.kt:
```diff
@@ -0,0 +1,93 @@
+    val formattingStatus: String,
+    val createdAt: Instant,
+    val updatedAt: Instant,
+)
```
レスポンス、リクエストは別ファイルに書き出して欲しいです。
理由としてはコントローラはこれからの実装でどんどん実装が膨れ上がっていくので、リクエスト、レスポンスが別ファイルに書き出されている方が可読性がよくて、ユースケース層での実装がしやすいからです。

soichiro1357533 replied:
レスポンス定義をコントローラから分離しました。MemoController.kt はコントローラ本体だけにしてます。

soichiro1357533 committed:
レスポンスとリクエストを分離 #20
